src/include/daos_errno.h

```diff
@@ -31,6 +31,14 @@
 extern "C" {
 #endif
 
+#ifdef __GURT_ERRNO_H__
+#ifdef DAOS_VERSION
+#error "daos_errno.h included after gurt/errno.h"
```

Comment:
no sure to understand what this error is for. The commit message says that it should be allowed, no?

Reply:
It's allowed for 3rd party include.   This error will trigger when we attempt to load it in wrong order in DAOS code (because we define DAOS_VERSION)